src/Dapr.Cryptography/Encryption/DaprEncryptionGrpcClient.cs

```diff
@@ -0,0 +1,188 @@
+        {
+            bufferedResult.Write(item.Span);
+        }
+
+        return bufferedResult.WrittenMemory;
+    }
+
+    /// <summary>
+    /// Encrypts a stream using the Dapr Cryptography encryption functionality.
+    /// </summary>
+    /// <param name="vaultResourceName">The name of the vault resource used by the operation.</param>
+    /// <param name="plaintextStream">The stream containing the bytes of the plaintext value to encrypt.</param>
+    /// <param name="keyName">The name of the key to use from the Vault for the encryption operation.</param>
+    /// <param name="encryptionOptions">Options informing how the encryption operation should be configured.</param>
+    /// <param name="cancellationToken">A <see cref="CancellationToken"/> that can be used to cancel the operation.</param>
+    /// <returns>An array of encrypted bytes.</returns>
+    [Obsolete(
+        "The API is currently not stable as it is in the Alpha stage. This attribute will be removed once it is stable.")]
+    public override async IAsyncEnumerable<ReadOnlyMemory<byte>> EncryptAsync(
+        string vaultResourceName,
+        Stream plaintextStream,
+        string keyName,
+        EncryptionOptions encryptionOptions,
+        [EnumeratorCancellation] CancellationToken cancellationToken = default)
+    {
+        ArgumentVerifier.ThrowIfNullOrEmpty(vaultResourceName, nameof(vaultResourceName));
+        ArgumentVerifier.ThrowIfNullOrEmpty(keyName, nameof(keyName));
+        ArgumentVerifier.ThrowIfNull(plaintextStream, nameof(plaintextStream));
+        ArgumentVerifier.ThrowIfNull(encryptionOptions, nameof(encryptionOptions));
+
+        var shouldOmitDecryptionKeyName =
+            string.IsNullOrWhiteSpace(encryptionOptions
+                .DecryptionKeyName); //Whitespace isn't likely a valid key name either
+
+        var encryptRequestOptions = new Autogenerated.EncryptRequestOptions
+        {
+            ComponentName = vaultResourceName,
+            DataEncryptionCipher = encryptionOptions.EncryptionCipher.GetValueFromEnumMember(),
+            KeyName = keyName,
+            KeyWrapAlgorithm = encryptionOptions.KeyWrapAlgorithm.GetValueFromEnumMember(),
+            OmitDecryptionKeyName = shouldOmitDecryptionKeyName
+        };
+
+        if (!shouldOmitDecryptionKeyName)
+        {
+            ArgumentVerifier.ThrowIfNullOrEmpty(encryptionOptions.DecryptionKeyName,
+                nameof(encryptionOptions.DecryptionKeyName));
+            encryptRequestOptions.DecryptionKeyName = encryptRequestOptions.DecryptionKeyName;
+        }
+
+        var grpcCallOptions = DaprClientUtilities.ConfigureGrpcCallOptions(typeof(DaprEncryptionClient).Assembly,
+            this.DaprApiToken, cancellationToken);
+        var duplexStream = Client.EncryptAlpha1(grpcCallOptions);
+
+        using var streamProcessor = new EncryptionStreamProcessor();
+        await streamProcessor.ProcessStreamAsync(plaintextStream, duplexStream, encryptRequestOptions,
```

Comment:
One interesting thing I noticed in the dapr Go code is that it does encryption/decryption on the full buffer in memory (`byte[]`) rather than streaming.

Reply:
I can't speak to what happens in the runtime, but the specification is that this is implemented using a bidirectional streaming connection.